src/Surfnet/StepupMiddleware/ApiBundle/Identity/Repository/RaSecondFactorRepository.php

```diff
@@ -0,0 +1,133 @@
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupMiddleware\ApiBundle\Identity\Repository;
+
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Query;
+use Surfnet\StepupMiddleware\ApiBundle\Identity\Command\SearchRaSecondFactorCommand;
+use Surfnet\StepupMiddleware\ApiBundle\Identity\Entity\RaSecondFactor;
+
+class RaSecondFactorRepository extends EntityRepository
+{
+    /**
+     * @param string $id
+     * @return RaSecondFactor|null
+     */
+    public function find($id)
+    {
+        /** @var RaSecondFactor|null $secondFactor */
+        $secondFactor = parent::find($id);
+
+        return $secondFactor;
+    }
+
+    /**
+     * @param string $identityId
+     * @return RaSecondFactor[]
+     */
+    public function findByIdentityId($identityId)
+    {
+        return parent::findBy(['identityId' => $identityId]);
+    }
+
+    /**
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity) The amount of if statements do not necessarily make the method
+     *                                               below complex or hard to maintain.
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     *
+     * @param SearchRaSecondFactorCommand $command
+     * @return Query
+     */
+    public function createSearchQuery(SearchRaSecondFactorCommand $command)
+    {
+        $queryBuilder = $this
+            ->createQueryBuilder('sf')
+            ->andWhere('sf.institution = :institution')
+            ->setParameter('institution', $command->institution);
+
+        if ($command->name) {
+            $queryBuilder->andWhere('sf.name LIKE :name')->setParameter('name', "%$command->name%");
+        }
+
+        if ($command->type) {
+            $queryBuilder->andWhere('sf.type = :type')->setParameter('type', $command->type);
+        }
+
+        if ($command->secondFactorId) {
+            $queryBuilder
+                ->andWhere('sf.secondFactorId = :secondFactorId')
+                ->setParameter('secondFactorId', $command->secondFactorId);
+        }
+
+        if ($command->email) {
+            $queryBuilder->andWhere('sf.email LIKE :email')->setParameter('email', "%$command->email%");
+        }
+
+        if ($command->status) {
+            $queryBuilder->andWhere('sf.status = :status')->setParameter('status', $command->status);
+        }
+
+        switch ($command->orderBy) {
+            case 'name':
+            case 'type':
+            case 'secondFactorId':
+            case 'email':
+                $queryBuilder->orderBy("sf.$command->orderBy", $command->orderDirection === 'desc' ? 'DESC' : 'ASC');
+                break;
+            case 'status':
+                $queryBuilder
+                    ->addSelect('
+                        CASE WHEN sf.status=:unverified THEN 0 ELSE
+                        CASE WHEN sf.status=:verified THEN 1 ELSE
+                        CASE WHEN sf.status=:vetted THEN 2 ELSE
+                        CASE WHEN sf.status=:revoked THEN 3 ELSE 4
```

Comment:
Iirc this should be written as
```sql
CASE sf.status
    WHEN :unverified THEN 0
    WHEN :verified THEN 1
    WHEN :vetted THEN 2
    WHEN :revoked THEN 3
    ELSE 4
END CASE
```
rather than nest case statements, or does the QueryBuilder not accept that?